packages/console/src/pages/ApplicationDetails/index.tsx

```diff
@@ -1,46 +1,149 @@
 import { Application } from '@logto/schemas';
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useForm } from 'react-hook-form';
 import { useTranslation } from 'react-i18next';
-import { useParams } from 'react-router-dom';
+import { useLocation, useParams } from 'react-router-dom';
 import useSWR from 'swr';
 
 import BackLink from '@/components/BackLink';
 import Button from '@/components/Button';
 import Card from '@/components/Card';
 import CopyToClipboard from '@/components/CopyToClipboard';
+import FormField from '@/components/FormField';
 import ImagePlaceholder from '@/components/ImagePlaceholder';
+import TabNav, { TabNavLink } from '@/components/TabNav';
+import TextInput from '@/components/TextInput';
 import { RequestError } from '@/swr';
 import { applicationTypeI18nKey } from '@/types/applications';
 
 import * as styles from './index.module.scss';
 
+type OidcConfig = {
+  authorization_endpoint: string;
+  userinfo_endpoint: string;
+  token_endpoint: string;
+};
+
 const ApplicationDetails = () => {
   const { id } = useParams();
+  const location = useLocation();
+
   const { data, error } = useSWR<Application, RequestError>(id && `/api/applications/${id}`);
+  const { data: oidcConfig, error: fetchOidcConfigError } = useSWR<OidcConfig, RequestError>(
+    '/oidc/.well-known/openid-configuration'
+  );
+  const isLoading = !data && !error && !fetchOidcConfigError;
+  const dataFetched = data && oidcConfig;
+
+  const { handleSubmit, register, reset } = useForm<Application>({
+    defaultValues: data,
+  });
+
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
-  const isLoading = !data && !error;
+
+  const isNormalSettings = !location.pathname.includes('advanced-settings');
+
+  const buildTabLink = (tabLinkName: string) => {
+    const index = location.pathname.lastIndexOf('/');
+
+    if (index < 0) {
+      return location.pathname;
```

Comment:
This looks weird. Should it be as follows?
```typescript
return `${location.pathname}/${tabLinkName}`;
```

Reply:
better use `new URL()` to add a slash or remove duplicate ones

Reply:
No. See the reply below.
https://github.com/logto-io/logto/pull/381#discussion_r825757803

Reply:
updated